### ISSUE TITLE
User registration and permissions upgrade

### ISSUE THREAD

**Issue by mosdevly** [adborden/ballot-api#6]
I kinda got ahead of myself juuuuuust a little. I started creating a simple "Registration" button on the admin page and set about upgrading roles and permissions. I don't imagine there will be dozens of people registering, but it would be nice to make this a bit more automated for the few who do. 

I figured with a permissions update we can make it so that approval is needed and we can put in place any other stabilizing measures (like throttles, or captcha, etc) to keep the spammers and trolls at bay.

What do you think?

**Comment by mosdevly** [comment]
Also I'll use this as a place to bounce ideas before submitting PRs. Never know who's already working on anything and I want to be respectful of everyones time and ideas.

**Comment by adborden** [comment]
Thanks for thinking about this. What users are you expecting to be registering themselves? Folks contributing ballot information or developers using the API?

Right now we're just trying to get the basics right. We're working with the @sfdoran from Oakland PEC to make sure the admin forms make sense for people familiar with ballot data. I think self-registration might be getting ahead of ourselves for now, but I definitely want to think about what roles/groups we need. I wrote up a [little wiki](https://github.com/caciviclab/ballot-api/wiki/Users-and-permissions) assuming we would be manually adding folks for now.

**Comment by mosdevly** [comment]
I had developers in mind. Right now if I go to the heroku site I can't login and there's no option for me to register.

**Comment by adborden** [comment]
Yeah, that's a bug. It's part of the swagger UI but it's not hooked up for anything.

**Comment by adborden** [comment]
Created https://github.com/caciviclab/ballot-api/issues/8
